src/apirequest.ts

```diff
@@ -188,6 +188,18 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
     parameters.mediaUrl = urlTemplate.parse(parameters.mediaUrl).expand(params);
   }
 
+  // Rewrite url if rootUrl is globally set
+  if (
+    parameters.context._options.rootUrl !== undefined &&
+    options.url !== undefined
+  ) {
+    const path = options.url.slice(parameters.context._options.rootUrl.length);
```

Comment:
I think we could probably get away with using some methods from Node.js' URL API, rather than a regex;

Could this use `url.resolve`?

Reply:
Good one! Didn't know this API.